flang/test/Driver/OpenMP/target-filtering.f90

```diff
@@ -0,0 +1,61 @@
+!RUN: %flang_fc1 -emit-llvm -fopenmp %s -o - | FileCheck %s --check-prefixes HOST,ALL
```

Comment:
Is this really a driver test? This patch doesn't add any driver functionality, so there should be no driver tests, right?

Looks like all you want is to verify the lowering to LLVM, but that's different to testing the driver. Please could you move this file? I mostly want to avoid @pr-subscribers-flang-driver being tagged when no driver changes are made. Thanks!

Reply:
@kiranchandramohan previously suggested `flang/test/Driver/OpenMP/` as a potential directory to keep lowering to llvm IR tests. Can you please suggest a better alternative for these tests?

Reply:
The right way to do LIT tests is to test a single stage at a time. You can use the following flow as guidance.
1. For tests lowering to HLFIR + FIR + OpenMP from source use flang/test/Lower/OpenMP.
2. For tests converting from FIR + OpenMP to LLVM dialect use flang/test/Fir/convert-to-llvm-openmp-and-fir.fir
3. For tests converting form LLVM + OpenMP or core-dialect+OpenMP to LLVM + OpenMP use mlir/test/Conversion/OpenMPToLLVM/
4. For test converting from LLVM + OpenMP to LLVM IR use mlir/test/Target/LLVMIR/openmp-*

Reply:
If you are testing a specific pass then the tests should go to flang/test/Transforms/. If there are many OpenMP related tests you can create a sub-directory there. But note that the test here should be from FIR+OpenMP to FIR+OpenMP.

Reply:
We do already have tests for the partial stages.

I felt that we should have some end-to-end tests for things that rely on multiple stages. If you feel like these are unnecessary then I can get rid of the test being added in this patch, and put up a separate patch to remove the remainder.

Reply:
You can write an RFC to have flang/test/llvm-ir or flang/test/end-to-end get approval and then add tests there.
You can also add end to end and execution tests in https://github.com/llvm/llvm-test-suite/tree/main/Fortran.

> If you feel like these are unnecessary

I did not say this.

Reply:
I've put up an RFC here https://discourse.llvm.org/t/rfc-flang-new-directory-for-adding-end-to-end-tests-for-lowering-to-llvm-ir-in-flang/74872.

Shall I let the `flang/test/Driver/OpenMP/target-filtering.f90` test be there for now in the Driver? Or I can remove it and add later based on the RFC.

Reply:
I've removed this test for now so that the patch may land. Once the RFC is settled, I'll add the test accordingly.